src/gui/addnewtorrentdialog.cpp

```diff
@@ -317,27 +315,18 @@ AddNewTorrentDialog::AddNewTorrentDialog(const BitTorrent::TorrentDescriptor &to
     if (!hasMetadata())
         m_ui->stopConditionComboBox->addItem(tr("Metadata received"), QVariant::fromValue(BitTorrent::Torrent::StopCondition::MetadataReceived));
     m_ui->stopConditionComboBox->addItem(tr("Files checked"), QVariant::fromValue(BitTorrent::Torrent::StopCondition::FilesChecked));
-    const auto stopCondition = m_torrentParams.stopCondition.value_or(session->torrentStopCondition());
-    if (hasMetadata() && (stopCondition == BitTorrent::Torrent::StopCondition::MetadataReceived))
-    {
-        m_ui->startTorrentCheckBox->setChecked(false);
-        m_ui->stopConditionComboBox->setCurrentIndex(m_ui->stopConditionComboBox->findData(QVariant::fromValue(BitTorrent::Torrent::StopCondition::None)));
-    }
-    else
-    {
-        m_ui->startTorrentCheckBox->setChecked(!m_torrentParams.addPaused.value_or(session->isAddTorrentPaused()));
-        m_ui->stopConditionComboBox->setCurrentIndex(m_ui->stopConditionComboBox->findData(QVariant::fromValue(stopCondition)));
-    }
-    m_ui->stopConditionLabel->setEnabled(m_ui->startTorrentCheckBox->isChecked());
-    m_ui->stopConditionComboBox->setEnabled(m_ui->startTorrentCheckBox->isChecked());
     connect(m_ui->startTorrentCheckBox, &QCheckBox::toggled, this, [this](const bool checked)
     {
         m_ui->stopConditionLabel->setEnabled(checked);
         m_ui->stopConditionComboBox->setEnabled(checked);
     });
 
     m_ui->comboTTM->blockSignals(true); // the TreeView size isn't correct if the slot does its job at this point
-    m_ui->comboTTM->setCurrentIndex(session->isAutoTMMDisabledByDefault() ? 0 : 1);
+
+    // Any UI elements that depend on m_torrentParams should be configured in this method to
+    // allow them to be updated if the params change after metadata download.
+    refreshAfterTorrentParamsChange();
```

Comment:
It's a bad name. It should be named after what it does, e.g., loadAddTorrentParams.

Reply:
Done